src/Chapter4.hs

```diff
@@ -629,7 +672,10 @@ Can you implement a monad version of AND, polymorphic over any monad?
 🕯 HINT: Use "(>>=)", "pure" and anonymous function
 -}
 andM :: (Monad m) => m Bool -> m Bool -> m Bool
-andM = error "andM: Not implemented!"
+andM ma mb = ma >>= \a -> 
+  if not a 
+  then pure False
+  else mb
```

Comment:
That's a totally correct solution! :+1: 

Though, I would suggest reducing some complexity from the code. You can avoid calling `not` and just swap the `if-then-else` branches. In general cases, I can see, how using `not` can be beneficial if one branch contains significantly more code, and may want to swap them for stylistic difference. But here it doesn't matter :slightly_smiling_face: